google/datalab/contrib/bigquery/commands/_bigquery.py

```diff
@@ -147,22 +149,19 @@ def _get_execute_parameters(load_task_config_name, bq_pipeline_transformation_co
     # The name of query for execution; if absent, we return None as we assume that there is
     # no query to execute
     if 'query' not in bq_pipeline_transformation_config:
-      if any(key in bq_pipeline_transformation_config for key in ['large', 'mode']):
-        raise Exception('Query is not specified, but at least one query option is.')
+      # TODO(rajivpb): Log this: 'Query is not specified, but at least one query option is.')
       return None
-
-    execute_task_config['query'] = bq_pipeline_transformation_config['query']
-
-    # TODO(rajivpb): There is no unit-test coverage of this.
-
-    # One of 'create' (default), 'append' or 'overwrite' for the destination table in BigQuery
-    execute_task_config['mode'] = bq_pipeline_transformation_config.get('mode', 'create')
-
+    query = utils.commands.get_notebook_item(bq_pipeline_transformation_config['query'])
+    execute_task_config['sql'] = query.sql
+    if 'table' in bq_pipeline_output_config:
```

Comment:
nit: If you have a default value for these keys, you could do `bq_pipeline_output_config.get('table', None)` for example, instead of the checks.

Reply:
If the table is not present, I actually do not want to set the key on execute_task_config, hence this check.